homeassistant/components/subaru/sensor.py

```diff
@@ -223,7 +223,7 @@ def native_value(self) -> None | int | float:
         if unit in LENGTH_UNITS:
             return round(unit_system.length(current_value, unit), 1)
 
-        if unit in PRESSURE_UNITS and unit_system == IMPERIAL_SYSTEM:
+        if unit in PRESSURE_UNITS and unit_system != METRIC_SYSTEM:
```

Comment:
This is possible without making changes to core UnitSystem, but I do not like that the logic is inverted. It can be dangerous if more options are added to the thing that is being compared